grafana-init/grafana.py

```diff
@@ -121,31 +122,32 @@ def create_dashboard_payload(json_path):
 
 
 def main():
-    logging.info('Opening a Grafana session...')
-    session = Session()
-    login(session)
-
-    if check_initialized(session):
-        logging.info('Grafana has already been initialized, skipping!')
-        return
-
-    logging.info('Attempting to add configured datasource...')
-    r = session.post('{url}/api/datasources'.format(url=GRAFANA_URL),
-                     json=create_datasource_payload())
-    logging.debug('Response: %r', r.json())
-    r.raise_for_status()
-
-    for path in sorted(glob.glob('{dir}/*.json'.format(dir=DASHBOARDS_DIR))):
-        logging.info('Creating dashboard from file: {path}'.format(path=path))
-        r = session.post('{url}/api/dashboards/db'.format(url=GRAFANA_URL),
-                         json=create_dashboard_payload(path))
+    for user in create_login_payload():
+        logging.info('Opening a Grafana session...')
+        session = Session()
+        login(session, user)
+
+        if check_initialized(session):
+            logging.info('Grafana has already been initialized, skipping!')
+            return
+
+        logging.info('Attempting to add configured datasource...')
+        r = session.post('{url}/api/datasources'.format(url=GRAFANA_URL),
+                         json=create_datasource_payload())
         logging.debug('Response: %r', r.json())
         r.raise_for_status()
 
-    logging.info('Ending session...')
-    session.get('{url}/logout'.format(url=GRAFANA_URL))
+        for path in sorted(glob.glob('{dir}/*.json'.format(dir=DASHBOARDS_DIR))):
+            logging.info('Creating dashboard from file: {path}'.format(path=path))
+            r = session.post('{url}/api/dashboards/db'.format(url=GRAFANA_URL),
+                             json=create_dashboard_payload(path))
+            logging.debug('Response: %r', r.json())
+            r.raise_for_status()
+
+        logging.info('Ending session...')
+        session.get('{url}/logout'.format(url=GRAFANA_URL))
 
-    logging.info('Finished successfully.')
+        logging.info('Finished successfully.')
```

Comment:
You should not increase indentation here, you want to print that message only when you went through all users.